gap/fp/freeband.gi

```diff
@@ -420,6 +420,323 @@ function(S)
   fi;
 end);
 
+InstallMethod(EqualInFreeBand, "for two strings",
+[IsString, IsString],
+function(w1_in, w2_in)
+  if IsEmpty(w1_in) or IsEmpty(w2_in) then
+    return IsEmpty(w1_in) and IsEmpty(w2_in);
+    # Necessary since [] is recognised as a string yet the below
+    # doesn't work with it.
+  fi;
+  w1_in := List(w1_in, IntChar);
+  w2_in := List(w2_in, IntChar);
+  return EqualInFreeBand(w1_in, w2_in);
+end);
+
+# This is an implementation of the algorithm described in Jakub Radoszewski
+# and Wojciech Rytter's paper 'Efficient Testing of Equivalence of Words
+# in a Free Idempotent Semigroup⋆'
+
+InstallMethod(EqualInFreeBand, "for two lists of positive integers",
+[IsList, IsList],
```

Comment:
```suggestion
[IsHomogeneousList, IsHomogeneousList],
```

Reply:
This way if the lists have holes, or contain things that are not in the same family, this method won't be selected. Reducing the checks that you should do at the start of the function itself.